### PR TITLE
Support for user control over number of threads for tensorflow backend

### PR DESCRIPTION
**Pull request by Epist**
Added support for user modification of the number of threads that the tensorflow backend uses to run its "intra_op_parallelism". This was done by making it so that the tensorflow backend respects the setting of the environment variable OMP_NUM_THREADS, which is usually used to set the number of threads that a BLAS package runs with. If the environment variable is not set, then tensorflow runs with its default behavior, which is to use the maximum number of threads that are accessible.

The reasons to limit the number of threads are twofold. First, one might wish to run multiple instances of keras concurrently on the same machine, which would be impeded by a process that consumes all of the available computational resources. Second, there are cases where the software actually runs faster on fewer cores, such as when the overhead for locking and communicating between parallel processes
scales faster than the computuational benefit of the parallel processes. This was observed on a machine with two 8-core (16 thread) Intel(R) Xeon(R) CPU E5-2660 0 @ 2.20GHz CPUs. Using more than 8 threads at a time resulted in a significant slowdown using the lstm_text_generation example. Using fewer threads in BLAS alone did not improve performance, and neither did decreasing the number of threads in tensorflow while leaving BLAS alone. It is because of this relationship that the number of threads available to tensorflow is set to track the number of threads available to BLAS. Additionally, this makes the behavior of the tensorflow backend equivalent to the behavior of the theano backend, which already adjusts its performance implicitly based on the setting of the OMP_NUM_THREADS environment variable.

In short, this improves the user control, performance, and  consistency of the keras package.

### CODE REVIEW

**Comment by Epist** [comment]
Not sure how this could cause a theano test to fail. The only file that was changed would not even be called with theano set as the backend.

**Comment by tboquet** [comment]
Hi, I think this change should be added in the `__init__.py` ([here](https://github.com/fchollet/keras/blob/master/keras/backend/__init__.py#L19-L29)) file and the `./keras/keras.json` file if this functionnality makes sense.
Your tests are failing because the code is not PEP8 compliant. You could take a look at the errors [here](https://travis-ci.org/fchollet/keras/jobs/107955300#L780-L789).

**Comment by fchollet** [comment]
Multiple issues with this PR. I merged a much simpler version: https://github.com/fchollet/keras/commit/20728c95fad32072470645c3c34c2044cdc1557b
